### DISCUSSION TITLE
Custom Materials: Explicit Assignment Issue and some Observations

### DISCUSSION THREAD

**Discussion by steffenitely** [NVlabs/sionna#660] [Q&A]
Hi everyone,

I’ve been working with custom materials in my simulation environment and ran into some unexpected issues while adjusting their parameters. To investigate, I set up a simple test scenario to observe how parameter changes in custom materials affect the amplitude of reflected paths. The setup includes a single transmitter (Tx) and receiver (Rx) placed at equal distances from two walls with custom materials. For this simulation, only the Line-of-Sight (LOS) and first-order reflections were considered. Here are a couple of key findings and areas of confusion:

1) Custom Material Parameters Require Explicit Reassignment to Take Effect

I noticed that altering the parameters of custom materials (such as permittivity or conductivity) did not immediately impact the path coefficient in my simulations. The path power remained unchanged unless I explicitly reassigned the custom material to the object it was associated with, even if the material was already assigned in the .xml file.

Even more confusion is caused by checking the radio material of the object using scene.get("object_name"). Here the material of the object is displayed which was assigned using blender and defined in the notebook beforehand. The correct paramteres are always shown; with or without the explicit assignment but then the simulation is only effected with the explicit assignment. 
This behavior can make simulations challenging, especially in larger scenes with multiple objects using different custom materials. Below are images showing the delay and power of the paths with and without explicit material reassignment:

Without explicit reassignment: (pic)
![image](https://github.com/user-attachments/assets/297be27f-ad2b-4d67-a53d-24a21eb7ad0c)

With explicit reassignment: (pic)
![image](https://github.com/user-attachments/assets/28d388d5-075b-49a2-aadf-b57a125188bb)

As you can see, while custom materials work, it’s counterintuitive that they need to be explicitly reassigned to take effect. This could complicate workflows in larger scenes with numerous custom materials.

It would be very helpful if the explicit assignment step could be skipped and defining is enough for a correct use of paramters when simulating

2) Scattering Coefficient’s Impact on Reflected Paths

I also observed that the scattering coefficient affects the power of reflected paths, which makes sense. However, I’d suggest adding a small note in the documentation as a reminder that the scattering coefficient impacts reflections even when scattering = False. This would help clarify the coefficient’s effect for other users who may not expect it.

Thanks for all help in advance and best regards

**Comment by faycalaa** [comment]
Hi @steffenitely,

Concerning 1., could you provide a minimal code snippet that reproduces the issue?

**Reply by steffenitely** [reply]
Hi @faycalaa, 
here is the zip file of the the test ground and the code to reproduce the issue: 
[MaterialTest.zip](https://github.com/user-attachments/files/17803128/MaterialTest.zip)


You will get different results for the paths depending if the cell
 
```py
obj1 = scene.get("Cube")
obj1.radio_material = "custom1" 

obj2 = scene.get("Cube_001")
obj2.radio_material = "transparent"
```

was run or not.

**Reply by steffenitely** [reply]
Hi @faycalaa 
I wanted to check in and ask if you were able to recreate the issue?

**Reply by HelmutWolfgang** [reply]
Has there been any news on this? I think this issue is still present.

**Reply by steffenitely** [reply]
Hi @HelmutWolfgang, 
sadly there were no updates to this issue, but i think it is also because of the new version of Sionna (1.0). As of Sionna 1.0.2 there is a bug that prohibits loading a scene with custom materials. Once that big has been solved i think we can try and see if the explicit assignment of materials is still needed.

**Reply by merlinND** [reply]
Hello,

Please see https://github.com/NVlabs/sionna/discussions/879, custom materials and built-in radio materials will be easier to handle in the next minor release.
